lib/src/modules/gates.dart

```diff
@@ -772,13 +772,20 @@ class ReplicationOp extends Module
 
   /// Constructs a ReplicationOp
   ///
-  /// The bit [bit] will be repeated over the [_width] as an output.
+  /// The signal [original] will be repeated over the [_multiplier] times as an
+  /// output.
+  /// Input [_multiplier] cannot be negative or zero, an exception will be
+  /// thrown, otherwise.
   /// [Module] is in-lined as SystemVerilog, it will use {width{bit}}
-  ReplicationOp(Logic bit, this._width)
-      : _inputName = 'input_${bit.name}',
-        _outputName = Module.unpreferredName('output_${bit.name}') {
-    addInput(_inputName, bit);
-    addOutput(_outputName, width: _width);
+  ReplicationOp(Logic original, this._multiplier)
+      : _inputName = 'input_${original.name}',
```

Comment:
I'm unsure if SV allows chaining of replication operators.  Did you run into issues making the input an `unpreferredName` when simulating the generated SV?  If we don't have a test that stresses this a bit (chaining replication operators with other in-line SV operations), we should.  It's better to make it "unpreferred" if we can.